pennylane/ops/qutrit/parametric_ops.py

```diff
@@ -306,10 +310,155 @@ def compute_matrix(theta, subspace=(0, 1)):  # pylint: disable=arguments-differ
+
+    grad_method = "A"
+    parameter_frequencies = [(0.5, 1)]
+
+    def generator(self):
+        if self.subspace == (0, 1):
+            return qml.s_prod(-0.5, qml.GellMann(wires=self.wires, index=3))
+
+        if self.subspace == (0, 2):
+            coeffs = [-0.25, -0.25 * np.sqrt(3)]
+            obs = [qml.GellMann(wires=self.wires, index=3), qml.GellMann(wires=self.wires, index=8)]
+            return qml.dot(coeffs, obs)
+
+        coeffs = [-0.25 * np.sqrt(3), 0.25]
+        obs = [qml.GellMann(wires=self.wires, index=8), qml.GellMann(wires=self.wires, index=3)]
+        return qml.dot(coeffs, obs)
+
+    def __init__(self, phi, wires, subspace=(0, 1), do_queue=True, id=None):
+        self._subspace = self.validate_subspace(subspace)
+        self._hyperparameters = {
+            "subspace": self._subspace,
+        }
+        super().__init__(phi, wires=wires, do_queue=do_queue, id=id)
+
+    @property
+    def subspace(self):
+        """The single-qutrit basis states which the operator acts on
+
+        This subspace determines which two single-qutrit basis states the operator acts on.
+        The remaining basis state is not affected by the operator.
+
+        Returns:
+            tuple[int]: subspace on which operator acts
+        """
+        return self._subspace
+
+    @staticmethod
+    def compute_matrix(theta, subspace=(0, 1)):  # pylint: disable=arguments-differ
+        r"""Representation of the operator as a canonical matrix in the computational basis (static method).
+
+        The canonical matrix is the textbook matrix representation that does not consider wires.
+        Implicitly, this assumes that the wires of the operator correspond to the global wire order.
+
+        .. seealso:: :meth:`~.TRZ.matrix`
+
+        Args:
+            theta (tensor_like or float): rotation angle
+            subspace (Sequence[int]): the 2D subspace on which to apply operation
+
+        Returns:
+            tensor_like: canonical matrix
+
+        **Example**
+
+        >>> qml.TRZ.compute_matrix(torch.tensor(0.5), subspace=(0, 2))
+        tensor([[0.9689-0.2474j, 0.0000+0.0000j, 0.0000+0.0000j],
+                [0.0000+0.0000j, 1.0000+0.0000j, 0.0000+0.0000j],
+                [0.0000+0.0000j, 0.0000+0.0000j, 0.9689+0.2474j]])
+        """
+        if qml.math.get_interface(theta) == "tensorflow":
+            theta = qml.math.cast_like(theta, 1j)
+        p = qml.math.exp(-1j * theta / 2)
+
+        shape = qml.math.shape(theta)
+        is_broadcasted = len(shape) != 0 and shape[0] > 1
+        # Construct identity matrices and cast to complex type
+        mat = (
+            qml.math.tensordot([1] * shape[0], qml.math.eye(3), axes=0)
+            if is_broadcasted
+            else qml.math.eye(3)
+        )
+        mat = qml.math.cast_like(mat, p)
+
+        slices = [(i, i) for i in subspace]
+        if is_broadcasted:
+            slices = [(Ellipsis, *s) for s in slices]
```

Comment:
Shouldn't this work for the non-broadcasting  case as well? 
```suggestion
        slices = [(Ellipsis, i, i) for i in subspace]
```